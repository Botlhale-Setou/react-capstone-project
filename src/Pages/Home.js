import React, { useState } from "react";
import { useSelector } from "react-redux";
import Country from "../Components/Country";

const Home = () => {
  const { countries } = useSelector((state) => state.countries);
  let countryDB = [];

  const [coDB, setCoDB] = useState(countries);

  const regionSwitch = (e) => {
    switch (e.target.value) {
      case "all":
        countryDB = Array.from(countries);
        setCoDB(countryDB);
        break;
      case "Northern":
        countryDB = countries.filter(
          (item) => item.region === "Northern Africa"
        );
        setCoDB(countryDB);
        break;
      case "Eastern":
        countryDB = countries.filter(
          (item) => item.region === "Eastern Africa"
        );
        setCoDB(countryDB);
        break;
      case "Southern":
        countryDB = countries.filter(
          (item) => item.region === "Southern Africa"
        );
        setCoDB(countryDB);
        break;
      case "Western":
        countryDB = countries.filter(
          (item) => item.region === "Western Africa"
        );
        setCoDB(countryDB);
        break;
      case "Middle":
        countryDB = countries.filter((item) => item.region === "Middle Africa");
        setCoDB(countryDB);
        break;

      default:
        break;
    }
  };

  return (
    <div className='home'>
      <label for='regionSelect'>Select region:</label>
      <select
        name='regionSelect'
        id='regionSelect'
        defaultValue={"all"}
        className='regionSelect'
        onChange={(e) => regionSwitch(e)}
      >
        <option value='all'>All (59)</option>
        <option value='Northern'>Northern (7)</option>
        <option value='Eastern'>Eastern (19)</option>
        <option value='Southern'>Southern (6)</option>
        <option value='Western'>Western (17)</option>
        <option value='Middle'>Middle (10)</option>
      </select>

      {coDB.map((item) => (
        <Country
          key={item.name}
          name={item.name}
          flag={item.flags.png}
          area={item.area}
        />
      ))}
    </div>
  );
};

export default Home;
