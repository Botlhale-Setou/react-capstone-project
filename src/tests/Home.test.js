import React from 'react';
import Renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../Redux/ConfigureStore';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import Home from '../Pages/Home';

describe('Home test', () => {
	test('Home render', () => {
		const tree = Renderer
      .create(
        <Provider store={store}>
          <Home />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
	});
});