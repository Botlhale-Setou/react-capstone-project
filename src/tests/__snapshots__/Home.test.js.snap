// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home test Home render 1`] = `
<div
  className="home"
>
  <label
    for="regionSelect"
  >
    Select region:
  </label>
  <select
    className="regionSelect"
    defaultValue="all"
    id="regionSelect"
    name="regionSelect"
    onChange={[Function]}
  >
    <option
      value="all"
    >
      All (59)
    </option>
    <option
      value="Northern"
    >
      Northern (7)
    </option>
    <option
      value="Eastern"
    >
      Eastern (19)
    </option>
    <option
      value="Southern"
    >
      Southern (6)
    </option>
    <option
      value="Western"
    >
      Western (17)
    </option>
    <option
      value="Middle"
    >
      Middle (10)
    </option>
  </select>
</div>
`;
